src/frontend.md

```diff
@@ -289,7 +292,43 @@ export function fakeTicket() {}
 
 ## TanStack Query
 
-1. Return the full response from a request
+1. Prefer use important defaults. Read more about defaults [here](https://tanstack.com/query/v4/docs/react/guides/important-defaults) and [here](https://www.notion.so/osome/Differences-and-Features-of-cacheTime-and-staleTime-parameters-f1f622beb2144b05a4e532e0324334dd)
+
+   > Why? It prevents some cache issues are usually difficult to troubleshoot and resolve. In addition, some settings provide a good user experience (for example, `refetchOnWindowFocus`)
+
+   ```typescript
+   // bad
+   const queryClient = new QueryClient({
+     defaultOptions: {
+       queries: {
+         refetchOnWindowFocus: false,
+         refetchOnMount: false,
+       },
+     },
+   });
+
+   // good
+   const queryClient = new QueryClient({
+     defaultOptions: {
+       queries: {
+         refetchOnWindowFocus: process.env.ENV === 'production',
```

Comment:
we'd better avoid different settings for local, stage and production as much as possible because it may lead to unexpected behavior

Reply:
I based this option [on the article](https://tkdodo.eu/blog/practical-react-query#the-defaults-explained)

>If you see a refetch that you are not expecting, it is likely because you just focused the window and React Query is doing a refetchOnWindowFocus, which is a great feature for production: If the user goes to a different browser tab, and then comes back to your app, a background refetch will be triggered automatically, and data on the screen will be updated if something has changed on the server in the meantime. All of this happens without a loading spinner being shown, and your component will not re-render if the data is the same as you currently have in the cache.
>During development, this will probably be triggered more frequently, especially because focusing between the Browser DevTools and your app will also cause a fetch, so be aware of that.

Reply:
BTW, `refetchOnWindowFocus: true` might be a bit of dangerous default.

Say, there's a document uploader on a page. Clicking it opens a system dialog and causes window to lose focus. When dialog is closed, window gets focus back and that causes refetch. In [some cases](https://reallyosome.atlassian.net/browse/NEWBORN-1277) that might cause inputs on the page to lose state.

Reply:
I think, [Such cases](https://reallyosome.atlassian.net/browse/NEWBORN-1277) are rare.
In most cases `refetchOnWindowFocus: true` is good, and we can set `refetchOnWindowFocus: false` if needed.

Reply:
> 

Yes, we can miss some bugs during development

Reply:
considering such bugs maybe `refetchOnWindowFocus: false` and enabling `refetchOnWindowFocus: true` on demand looks like better option

moreover, right now websome has `refetchOnWindowFocus: false` and after we enable it and share query client between all MFEs, i am not sure how many bugs we will get

so i vote for `refetchOnWindowFocus: false` as a safe default option and enabling it on demand